Fix handling of multiple filters in podman ps

mheon opened the pull request:
Docker expects multiple filters to be passed with multiple uses of the --filter flag (e.g. --filter=label=a=b --filter=label=c=d) and not a single comma-separated list of filters as we expected. Convert to the Docker format, and make some small cleanups to our handling of filters along the way.

Fixes #1341

mheon commented:
Note to self: manpages and bash completions probably also need updating

TomSweeneyRedHat commented on test/e2e/ps_test.go:
```diff
@@ -181,6 +181,18 @@ var _ = Describe("Podman ps", func() {
+		Expect(result.ExitCode()).To(Equal(0))
+		Expect(result.OutputToStringArray()[0]).To(Equal(fullCid))
+	})
+
```
I think this new test might have succeeded with the old behavior.  It might be good to create a second container with key1==value2 and then make sure you only get one container back.

EmilienM commented:
I tested the PR and it fixed my bug. Thanks again @mheon !

mheon commented:
Fixed the test and added another container to make sure the match logic for multiple filters is logical-AND, and updated the manpage. This should be ready now. CI seems to be having its usual issues though.

mheon commented:
@rhatdan @umohnani8 @baude @vrothberg PTAL

umohnani8 commented:
LGTM

mheon commented:
@rh-atomic-bot r=umohnani8
